scripts/zkai_d128_snark_ivc_statement_receipt_gate.py

```
@@ -0,0 +1,816 @@
+    if not payload["all_mutations_rejected"] or not all(case.get("rejected") is True for case in cases):
+        raise D128SnarkReceiptError("not all SNARK receipt mutations rejected", layer="mutation_suite")
+    by_name = {case.get("mutation"): case for case in cases}
+    expect_equal(set(by_name), EXPECTED_MUTATION_SET, "case mutation set")
+    for mutation, surface in EXPECTED_MUTATION_INVENTORY:
+        case = by_name[mutation]
+        expect_equal(case.get("surface"), surface, f"surface for {mutation}")
+        if case.get("mutated_accepted") is True:
+            raise D128SnarkReceiptError(f"mutation accepted: {mutation}", layer="mutation_suite")
+    metrics = require_object(payload["receipt_metrics"], "receipt metrics")
+    if metrics.get("proof_size_bytes") != artifact_path("proof").stat().st_size:
+        raise D128SnarkReceiptError("proof size metric mismatch", layer="receipt_metrics")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)